pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessorTest.java

```diff
@@ -500,6 +494,123 @@ public void testEnableDictAndOtherIndexesMV()
         false, 13, null, false, DataType.INT, 106688);
   }
 
+  @Test
+  public void testSimpleDisableDictionary()
+      throws Exception {
+    // TEST 1. Check running forwardIndexHandler on a V1 segment. No-op for all existing dict columns.
+    constructV1Segment(Collections.emptyList(), Collections.emptyList(), Collections.emptyList());
+    checkForwardIndexCreation(EXISTING_STRING_COL_DICT, 9, 4, _schema, false, true, false, 26, null, true, 0,
+        DataType.STRING, 100000);
+    validateIndex(ColumnIndexType.FORWARD_INDEX, COLUMN10_NAME, 3960, 12, _schema, false, true, false, 0, true, 0, null,
+        false, DataType.INT, 100000);
+
+    // Convert the segment to V3.
+    new SegmentV1V2ToV3FormatConverter().convert(_indexDir);
+
+    // TEST 2: Disable dictionary for EXISTING_STRING_COL_DICT.
```

Comment:
Add one test for fixed width type as well ?

Reply:
Can we also test bulk mode (disable on more than one column or a combination of disable and enable) if not already being tested ?

Reply:
We already have a test for int column column10 below. Does that work?

Yes, we do have bulk operations test in both ForwardIndexHandlerReloadQueriesTest and ForwardIndexHandlerTest files.

Reply:
Query based tests will certainly help to check correctness before and after. I think you already have that. May be we can come up with a test query where range predicate is full match and thus results in reading the entire range index ?

Reply:
My comment crossed with yours (didn't see your response before). I am good with the tests we have. Might want to consider adding a test suggested in previous comment ^^

Reply:
Sounds good. Thanks